apisix/plugins/proxy-rewrite.lua

```diff
@@ -345,8 +346,21 @@ function _M.rewrite(conf, ctx)
         else
             ctx.var.upstream_uri = upstream_uri
         end
+    else
+        ctx.var.upstream_uri = upstream_uri
     end
 
+    if conf.use_real_request_uri_unsafe and conf.regex_uri then
+        local index
+        if separator_escaped then
+            index = str_find(upstream_uri,"?")
+        end
+        if index then
+            upstream_uri = sub_str(upstream_uri, 1, index - 1)
+                           ..sub_str(upstream_uri,index)
+        end
+        ctx.var.upstream_uri = upstream_uri
+    end
```

Comment:
Why do you add these code?

Reply:
To make regex_uri work with real_request_uri field. This is also what @AlinsRan had done in the diff shared.

Reply:
You're right, in the if block above this when the safe encoding was on then this logic was added to first remove the `?` and then readd after safely encoding. When we're skipping the safe encoding then we don't need to do any of that but I added it by mistake. Removed this.